pkg/server/endpoints/node/handler.go

```diff
@@ -595,6 +596,35 @@ func (h *Handler) getAttestResponse(ctx context.Context,
 	return &node.AttestResponse{SvidUpdate: svidUpdate}, nil
 }
 
+func (h *Handler) getDownstreamEntry(ctx context.Context, callerID string) *common.RegistrationEntry {
+	ds := h.c.Catalog.DataStores()[0]
+	response, err := ds.ListRegistrationEntries(ctx, &datastore.ListRegistrationEntriesRequest{
+		BySpiffeId: &wrappers.StringValue{
+			Value: callerID,
+		},
+	})
+
+	if err != nil {
+		return nil
+	}
+
+	regEntriesMap := make(map[string]*common.RegistrationEntry)
+	for _, entry := range response.Entries {
+		regEntriesMap[entry.SpiffeId] = entry
```

Comment:
There can be many entries with the same SPIFFE ID. Storing them in a map like this will result in last-record-wins. Instead, we should loop through the entries looking for one that has Downstream set to true... when we find it, return... then at the end of the function, we would have something like:
```go
return nil, fmt.Errorf("%v is not an authorized downstream workload", callerID)
```

I think we have some things we need to sort out long-term with regards to multiple entries with the same SPIFFE ID, but for now, this should suffice.

Reply:
Fixed as per the suggestion .